web/docs/guides/with-redwoodjs.mdx

```diff
@@ -0,0 +1,746 @@
+
+[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https%3A%2F%2Fgithub.com%2Fsupabase%2Fsupabase%2Ftree%2Fmaster%2Fexamples%2Freact-user-management&project-name=supabase-user-management&repository-name=supabase-user-management&demo-title=Supabase%20User%20Management&demo-description=An%20example%20web%20app%20using%20Supabase%20and%20Next.js&demo-url=https%3A%2F%2Fsupabase-react-user-management.vercel.app&demo-image=https%3A%2F%2Fi.imgur.com%2FZ3HkQqe.png&integration-ids=oac_jUduyjQgOyzev1fjrW83NYOv&external-id=nextjs-user-management)
+
+Clicking this button the application will:
+
+- Launch and prepare the Postgres database in Supabase.
+- Launch the app in Vercel.
+- Fork the example into your own GitHub account.
+- Prepare the deployed application with all the necessary environment variables.
+
+If you want to do it yourself, let's get started: -->
+
+## About RedwoodJS
+
+A Redwood application is split into two parts: a frontend and a backend. This is represented as two node projects within a single monorepo.
+
+The frontend project is called **web** and the backend project is called **api**. For clarity, we will refer to these in prose as **"sides"**, i.e. the "web side" and the "api side".
+They are separate projects because code on the web side will end up running in the user's browser while code on the api side will run on a server somewhere.
+
+> Important: When this guide refers to "API", that means the Supabase API and when it refers to "api side", that means the RedwoodJS api side.
+
+The **api side** is an implementation of a GraphQL API. Your business logic is organized into "services" that represent their own internal API and can be called both from external GraphQL requests and other internal services.
+Redwood can automatically connect your internal services with Apollo, reducing the amount of boilerplate you have to write.
+Your services can interact with a database via Prisma's ORM, and Prisma's migration tooling provides first-class migrations that take the pain out of evolving your database schema.
+
+The **web side** is built with React. Redwood's router makes it simple to map URL paths to React "Page" components (and automatically code-split your app on each route).
+Pages may contain a "Layout" component to wrap content. They also contain "Cells" and regular React components.
+Cells allow you to declaratively manage the lifecycle of a component that fetches and displays data.
+
+> FYI: For the sake of consistency with the other framework Quickstart examples, we'll build a RedwoodJS a little differently than normal.
+> We **_won't use_** use Prisma to connect to the Supabase Postgres database or [Prisma migrations](https://redwoodjs.com/docs/cli-commands#prisma-migrate) as one typically might in a Redwood app.
+> Instead, we'll rely on the Supabase client to do some of the work on the **web** side and use the client again on the **api** side to do data fetching as well.
```

Comment:
Your idea for a second guide sounds great (I don't use Redwood, but am definitely interested in using Prisma with Supabase's auth)! 👍

Reply:
> Your idea for a second guide sounds great (I don't use Redwood, but am definitely interested in using Prisma with Supabase's auth)! 👍

Thanks @amosbastian lots of redwood devs are using Supabase right now with Prisma.

If you are interested in RW, the Tutorial is a great place to start: https://learn.redwoodjs.com/docs/tutorial/prerequisites

And it can support Supabase auth as well.

The key things to know are:

* Connection Pooling w/ Serverless and how Prisma deals with the Migrations: https://community.redwoodjs.com/t/guide-migrations-connection-pool-urls-supabase/2505
* Prisma cannot help with RLS at the moment

I'd like to do two other RW guides:

* RW + Supabase + Prisma Client on api side
* RW + Supabase without Prisma but just using Supabase SDK on api side

Reply:
Did you ever end up writing this 2nd guide @dthyresson?

Reply:
> Did you ever end up writing this 2nd guide @dthyresson?

HI @amosbastian. No, not yet. Namely because the current Quick Start is heavy on auth and storage, but a little light on data fetching and rendering. Instead, one showcase / QuickStart of RedwoodJS with Supabase is simply to take the [RedwoodJS tutorial](https://redwoodjs.com/docs/tutorial/foreword) -- but use Supabase as the Postgres database with PgBouncer turned on.

Instead, I am thinking of porting over https://github.com/supabase-community/supabase-graphql-example to RedwoodJS instead. For both those scenarios: Prisma Client / Supabase client on api side.

Redwood launches v1 this week! https://v1launchweek.redwoodjs.com So, check it out :)